WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -34,9 +34,18 @@ class PostCoordinator: NSObject {
             post.deleteRevision()
         }
 
+        if post.hasFailedMedia {
+            for media in post.media {
+                guard media.remoteStatus == .failed else {
+                    continue
+                }
+                mediaCoordinator.retryMedia(media)
+            }
+        }
+
         change(post: post, status: .pushing)
 
-        if mediaCoordinator.isUploadingMedia(for: post) {
+        if mediaCoordinator.isUploadingMedia(for: post) || post.hasFailedMedia {
```

Comment:
At first, I thought that `post.hasFailedMedia` is no longer necessary since we restart all the failed uploads. But it looks like we update `media.remoteStatus` in a block later on so we can't “catch” that change in here. What a shame.

Reply:
Exactly.

Unfortunately, I wasn't able to come up with a test case to ensure this behavior. If you have any ideas on how to make that explicit, please let me know.